modules/misc/gtk.nix

```diff
@@ -174,31 +232,41 @@ in {
     } // optionalAttrs (cfg.theme != null) { gtk-theme = cfg.theme.name; }
       // optionalAttrs (cfg.iconTheme != null) {
         icon-theme = cfg.iconTheme.name;
+      } // optionalAttrs (cfg.cursorTheme != null) {
+        cursor-theme = cfg.cursorTheme.name;
+      } // optionalAttrs
+      (cfg.cursorTheme != null && cfg.cursorTheme.size != null) {
+        cursor-size = cfg.cursorTheme.size;
       };
 
     optionalPackage = opt:
       optional (opt != null && opt.package != null) opt.package;
   in {
-    home.packages = optionalPackage cfg.font ++ optionalPackage cfg.theme
-      ++ optionalPackage cfg.iconTheme;
+    home.packages = concatMap optionalPackage [
+      cfg.font
+      cfg.theme
+      cfg.iconTheme
+      cfg.cursorTheme
+    ];
 
     home.file.${cfg2.configLocation}.text =
-      concatMapStrings (l: l + "\n") (mapAttrsToList formatGtk2Option ini)
+      concatStringsSep "\n" (mapAttrsToList formatGtk2Option gtkIni)
```

Comment:
There's a small difference between the two (when `gtkIni` isn't empty anyway), in that the previous form ends with a newline.  Which might be unnecessary sometimes, but also may be important in some settings re:produced file or at nix level if this value is appended to (as is done here with `cfg2.extraConfig`).
(mentioning because in PR text it seemed to suggest this was just intended as just a simplification not a behavior change)

Reply:
You are right, I can't believe I missed this. The missing newline will cause the first line of `cfg2.extraConfig` to be concatenated on the same line as the last line of the formatted `gtkIni`. `concatStringsSep` reserves space for the list of strings anyways so the change doesn't introduces any significant performance benefit. I will fix this shortly.